R/summary.rspec.R

```diff
@@ -158,21 +160,22 @@
 #' 13- Smiseth, P., J. Ornborg, S. Andersson, and T. Amundsen. 2001. Is male plumage reflectance
 #' correlated with paternal care in bluethroats? Behavioural Ecology 12:164-170.
 #'
-summary.rspec <- function(object, subset = FALSE, wlmin = NULL, wlmax = NULL, ...) {
+summary.rspec <- function(object, subset = FALSE, lim = c(300, 700), wlmin = NULL, wlmax = NULL, ...) {
   chkDots(...)
 
   wl <- isolate_wl(object, keep = "wl")
 
-  # Set WL min & max
-  lambdamin <- max(wlmin, min(wl))
-  lambdamax <- min(wlmax, max(wl))
-
-  if (!is.null(wlmin) && lambdamin > wlmin) {
-    stop("wlmin is smaller than the range of spectral data")
-  }
-  if (!is.null(wlmax) && lambdamax < wlmax) {
-    stop("wlmax is larger than the range of spectral data")
-  }
+  # Deprecate wlmin/wlmax
+  if (any(!is.null(wlmin), !is.null(wlmax))) {
+    warning("Arguments 'wlmin' and 'wlmax' are deprecated in favour of the single 'lim' argument, and will be removed in a future version.")
+    lim = c(wlmin, wlmax)
+  } 
+  
+  if (min(lim) < min(wl) || max(lim) > max(wl))
+    stop("Specified wavelemgth range exceeds that of the spectral data. Check the 'lim' argument.")
+  
+  lambdamin <- max(min(lim), min(wl))
+  lambdamax <- min(max(lim), max(wl))
 
   # Restrict to range of wlmin:wlmax
   object <- object[which(wl == lambdamin):which(wl == lambdamax), ]
```

Comment:
I didn't pay attention until now but this should probably be updated as well to allow `lim` values that don't necessarily match actually `wl` values (cf what we do below for S1)